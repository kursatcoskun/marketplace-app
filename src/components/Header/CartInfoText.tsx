import Text, { TextProps } from "antd/lib/typography/Text";
import React from "react";
import styled from "styled-components";

export const CartInfoText: React.FunctionComponent<TextProps> = styled(Text)`
  color: white;
  margin-left: 3px;
`;
