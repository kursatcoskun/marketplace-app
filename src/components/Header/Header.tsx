import styled from "styled-components";

export interface HeaderProps {}

const Header = styled.header`
  display: flex;
  overflow: hidden;
  align-items: center;
  background: #1ea4ce;
  color: white;
  font-family: Helvetica;
  font-weight: 300;
  height: 76px;
`;

export default Header;
