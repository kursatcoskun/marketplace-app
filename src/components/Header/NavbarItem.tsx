import styled from "styled-components";

const NavbarItem = styled.div`
  vertical-align: middle;
  background: #147594;
  height: -webkit-fill-available;
  padding: 12px 16px;
  margin-right: 7%;
  width: 129px;
  box-sizing: content-box;
  justify-content: center;
  justify-items: center;
  display: flex;
  align-items: center;
`;

export default NavbarItem;
