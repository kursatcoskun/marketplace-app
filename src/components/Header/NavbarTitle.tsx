import styled from "styled-components";

const NavbarTitle = styled.div`
  display: flex;
  justify-content: center;
  align-items: center;
  margin-left: auto;
  margin-right: auto;
  font-size: 150%;
  padding: 12px 16px;
`;

export default NavbarTitle;
