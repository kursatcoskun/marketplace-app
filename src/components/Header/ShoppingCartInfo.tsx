import { ShoppingCartOutlined } from "@ant-design/icons";
import { CartInfoText } from "./CartInfoText";
import { CenteredDiv } from "./CenteredDiv";
export interface ShoppingCartInfoProps {
  cartTotal: string | number;
}

const ShoppingCartInfo: React.FunctionComponent<ShoppingCartInfoProps> = (
  props
) => {
  return (
    <CenteredDiv>
      <ShoppingCartOutlined style={{ fontSize: "200%" }} />
      <CartInfoText>&#8378;{props.cartTotal}</CartInfoText>
    </CenteredDiv>
  );
};

export default ShoppingCartInfo;
