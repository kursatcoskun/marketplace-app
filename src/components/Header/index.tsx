import Header from "./Header";
import NavbarTitle from "./NavbarTitle";
import NavbarItem from "./NavbarItem";
import ShoppingCartInfo from "./ShoppingCartInfo";

export interface HeaderLayoutProps {
  title: String;
}

const HeaderLayout: React.FunctionComponent<HeaderLayoutProps> = (props) => {
  return (
    <Header>
      <NavbarTitle>{props.title}</NavbarTitle>
      <NavbarItem>
        <ShoppingCartInfo />
      </NavbarItem>
    </Header>
  );
};

export default HeaderLayout;
