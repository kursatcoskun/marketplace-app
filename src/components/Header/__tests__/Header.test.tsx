// MyComponent.test.js
import { shallow } from "enzyme";
import React from "react";
import Header from "../Header";
describe("CartInfoTextComponent", () => {
  it("should render Header component", () => {
    const wrapper = shallow(<Header />);
  });

  it("should render initial layout", () => {
    const component = shallow(<Header />);
    expect(component.getElements()).toMatchSnapshot();
  });
});
