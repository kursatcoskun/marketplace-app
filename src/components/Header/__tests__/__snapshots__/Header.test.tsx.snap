// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CartInfoTextComponent should render initial layout 1`] = `
Array [
  <header
    className="sc-bdnxRM fPdsWd"
  />,
]
`;
